src/AggregateFunctions/AggregateFunctionNull.cpp

```diff
@@ -108,14 +108,14 @@ class AggregateFunctionCombinatorNull final : public IAggregateFunctionCombinato
         {
             if (return_type_is_nullable)
             {
-                return std::make_shared<AggregateFunctionNullVariadic<true, true, true>>(nested_function, arguments, params);
+                return std::make_shared<AggregateFunctionNullVariadic<true, true>>(nested_function, arguments, params);
             }
             else
             {
                 if (serialize_flag)
-                    return std::make_shared<AggregateFunctionNullVariadic<false, true, true>>(nested_function, arguments, params);
+                    return std::make_shared<AggregateFunctionNullVariadic<false, true>>(nested_function, arguments, params);
                 else
-                    return std::make_shared<AggregateFunctionNullVariadic<false, true, false>>(nested_function, arguments, params);
```

Comment:
I am just wondering why the second template argument `serialize_flag` for `AggregateFunctionNullVariadic` is set to true when the local variable `serialize_flag` is **false**.

Reply:
It’s a bug that already existed before the change. Unfortunately fixing it is a breaking change and not worth it. There is a comment in the code now: https://github.com/ClickHouse/ClickHouse/blob/f2bad692e2835c58807daba0d4fa21eb4f440c1d/src/AggregateFunctions/Combinators/AggregateFunctionNull.cpp#L161